test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Discovery/TypeEnumeratorTests.cs

```diff
@@ -101,13 +103,12 @@ public void GetTestsShouldReturnBaseTestMethodsInSameAssembly()
 
             var tests = typeEnumerator.Enumerate(out this.warnings);
 
-            var methodCount = typeof(DummyDerivedTestClass).GetMethods(BindingFlags.Instance | BindingFlags.Public).Count(m => m.DeclaringType.Assembly == typeof(DummyDerivedTestClass).Assembly);
             Assert.IsNotNull(tests);
-            Assert.AreEqual(methodCount, tests.Count);
+            Assert.IsTrue(tests.Any(t => t.TestMethod.Name == "BaseTestMethod"));
```

Comment:
>Assert.IsTrue(tests.Any(t => t.TestMethod.Name == "BaseTestMethod")); [](start = 12, length = 69)

Do add a message on assertion failure so it is easy to identify why this test failed.